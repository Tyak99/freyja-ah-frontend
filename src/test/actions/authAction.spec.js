/* eslint-disable arrow-parens */
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import nock from 'nock';
import jwtDecode from 'jwt-decode';
import {
  registerUser,
  setCurrentUser,
} from '../../../store/actions/authActions';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const mockUser = {
  firstName: 'Ochowo',
  lastName: 'Jones',
  userName: 'gr',
  email: 'princess@gmail.com',
  password: 'password',
  confirmPassword: 'password',
};

const token =
  'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VySWQiOjI0LCJlbWFpbCI6InByaW5jZXNzQGdtYWlsLmNvbSIsInVzZXJOYW1lIjoiZ3IiLCJpYXQiOjE1NjI1Mjg2NTcsImV4cCI6MTU2MzEzMzQ1N30.oWthtPvSh-zz4RwgHZsJtdxpjhHlUKix0oK1I9nqkOA';

const decoded = jwtDecode(token);
describe('authAction', () => {
  let store;
  beforeEach(() => {
    store = mockStore({});
  });
  afterEach(() => {
    // clear all HTTP mocks after each test
    nock.cleanAll();
  });

  it('should decode user token when registerUser is successful', done => {
    nock('https://freyja-ah-backend.herokuapp.com')
      .post('/api/users')
      .reply(201, {
        message: 'user registration was successful',
        user: {
          id: 24,
          firstName: 'ochowo',
          lastName: 'Jones',
          email: 'princess@gmail.com',
          userName: 'gr',
        },
        token:
          'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VySWQiOjI0LCJlbWFpbCI6InByaW5jZXNzQGdtYWlsLmNvbSIsInVzZXJOYW1lIjoiZ3IiLCJpYXQiOjE1NjI1Mjg2NTcsImV4cCI6MTU2MzEzMzQ1N30.oWthtPvSh-zz4RwgHZsJtdxpjhHlUKix0oK1I9nqkOA',
      });

    store.dispatch(registerUser(mockUser)).then(() => {
      expect(store.getActions()).toMatchSnapshot();
      store.dispatch(setCurrentUser(decoded));
      expect(store.getActions()).toMatchSnapshot();
      done();
    });
  });
});
