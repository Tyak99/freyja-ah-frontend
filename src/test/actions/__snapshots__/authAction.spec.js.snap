// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`authAction should decode user token when registerUser is successful 1`] = `
Array [
  Object {
    "payload": Object {
      "email": "princess@gmail.com",
      "exp": 1563133457,
      "iat": 1562528657,
      "userId": 24,
      "userName": "gr",
    },
    "type": "SET_CURRENT_USER",
  },
  Object {
    "payload": [TypeError: Cannot read property 'push' of undefined],
    "type": "GET_ERRORS",
  },
]
`;

exports[`authAction should decode user token when registerUser is successful 2`] = `
Array [
  Object {
    "payload": Object {
      "email": "princess@gmail.com",
      "exp": 1563133457,
      "iat": 1562528657,
      "userId": 24,
      "userName": "gr",
    },
    "type": "SET_CURRENT_USER",
  },
  Object {
    "payload": [TypeError: Cannot read property 'push' of undefined],
    "type": "GET_ERRORS",
  },
  Object {
    "payload": Object {
      "email": "princess@gmail.com",
      "exp": 1563133457,
      "iat": 1562528657,
      "userId": 24,
      "userName": "gr",
    },
    "type": "SET_CURRENT_USER",
  },
]
`;
